core/modules/file/file.install

```diff
@@ -1199,3 +1211,24 @@ function file_update_1008() {
   }
   db_create_table('cache_entity_file', $table);
 }
+
+/**
+ * Ensures that the "no results" text in 'file_admin' view has a valid format.
+ */
+function file_update_1009() {
+  // Back in file_update_1000() we created this view containing 'filtered_html'
+  // as the "no results" text format, but if we were upgrading from a site that
+  // didn't contain that format, it results in an error. Here we ensure that the
+  // format for that text is a valid format and if it isn't, use the fallback
+  // format.
+  $view = views_get_view('file_admin');
+  if ($view) {
```

Comment:
We need to check first if this view has been modified by site admins before we change it programatically. Can you pull its storage state first, wrap this in an "if default" check?  I believe there are some examples for other view updates (home page promoted view, for example) where you can see how this is done.

Reply:
Sorry, maybe I'm missing something. This update only changes the view if it has an invalid text format for the "No results" text (whether or not it has been modified by the site admin). So, IIUC, you're saying that if the View was modified by a site admin and has an invalid format, we should leave the invalid format alone? That we'd only fix the invalid format if the View hasn't been modified?

That seems like a less-desirable behavior. If the text format is invalid for any reason, surely using the fallback format is an improvement over the invalid format.

And it seems like a pretty common use case might be that the site admin has made a change elsewhere in the View without noticing that the "No results" format was invalid, and it would still be a good thing to correct the invalid format, even if they've modified the View.

Is there any use case where leaving the text format invalid is preferable to it being the fallback format?

Reply:
Updated to check all displays of that view and fix if necessary.

Reply:
You were right the first time. 

> So, IIUC, you're saying that if the View was modified by a site admin and has an invalid format, we should leave the invalid format alone? That we'd only fix the invalid format if the View hasn't been modified?

Yes, that's what I meant. This is the process we've used for all other config updates. If the [config thing] has been modified in any way from its default state, we should not update it in an update hook. We do this because it's possible that the [config thing] has been modified in some way that would make our change cause other problems, and since there are an infinite number of changes that we can't realistically test for, it's not worth worrying about the edgecases where the view has been modified. Instead, we catch the majority of cases where we know the change is safe to make (where the view is unchanged) and disregard any modified config.

Reply:
This point was discussed in the issue (and I think is resolved), so I'm taking the liberty of marking the PR conversation resolved.